File: Runs/Java_repo_codegemma:7b/Test_3.java
public class Test_3 {

    public static void main(String[] args) {
            union {
                float f;
                int i;
            } c;
    
            c.f = 123.456f;
            System.out.printf("Float: %.2f, Int bits: %d\n", c.f, c.i);
    
            c.i = 1098765432;
            System.out.printf("Int: %d, Float bits: %.2f\n", c.i, c.f);
        }

}

File: C_repo/test_3.c
#include <stdio.h>

union Converter {
    int i;
    float f;
};

int main() {
    union Converter c;
    
    // Write as float, read as int
    c.f = 123.456f;
    printf("Float: %f, Int bits: %d\n", c.f, c.i);

    // Write as int, read as float
    c.i = 1098765432;
    printf("Int: %d, Float bits: %f\n", c.i, c.f);
    
    return 0;
}